pallets/contracts/src/lib.rs

```diff
@@ -419,6 +426,10 @@ decl_module! {
         pub fn change_template_meta_url(origin, code_hash: CodeHash<T>, new_url: Option<MetaUrl>) -> DispatchResult {
             // Ensure whether the extrinsic is signed & validate the `code_hash`.
             let (did, _) = Self::ensure_signed_and_template_exists(origin, code_hash)?;
+            // Ensure URL is limited in length.
+            if let Some(url) = &new_url {
+                ensure_string_limited::<T>(url)?;
+            }
```

Comment:
Did you miss this one? 
```suggestion
            ensure_opt_string_limited::<T>(new_url.as_ref())?;
```